File: C/Symmetric_Tree.c
bool isSymmetric(struct TreeNode* root) {
    if(!root) return true;
    bool help(struct TreeNode* l, struct TreeNode* r){
        if(!l && !r) return true;
        if(l && r && l->val != r->val) return false;
        if(!l || !r) return false;
        return help(l->left, r->right) && help(l->right, r->left);
    };
    return help(root->left, root->right);
}

File: Dart/Symmetric_Tree.dart
import './_DATATYPES.dart';

bool isSymmetric(TreeNode root) {
  bool symmetricHelper(TreeNode left, TreeNode right) {
    if (left == null && right == null) return true;
    if (left == null || right == null) return false;
    return left.val == right.val ? true : false;
  }
  return symmetricHelper(root.left, root.right);
}

main(List<String> args) {
  TreeNode a = TreeNode(5);
  a.left = TreeNode(3);
  a.right = TreeNode(3);

  print(isSymmetric(a));
}